variables.tf

```diff
@@ -163,10 +197,20 @@ variable "security_group_rules" {
   }
 }
 
-##############################################################################
-
 variable "user_data" {
   description = "User data to initialize BMS deployment"
   type        = string
   default     = null
 }
+
+variable "enable_secure_boot" {
+  description = "Indicates whether secure boot is enabled. If enabled, the image must support secure boot or the server will fail to boot."
+  type        = bool
+  default     = false
+}
+
+variable "tpm_mode" {
```

Comment:
If tpm_mode can only have certain string values, you may want to have a validation here to verify that the value passed is actually a valid string.

Also note that someone could pass null into this variable.